micrometer-core/src/main/java/io/micrometer/core/instrument/MeterRegistry.java

```diff
@@ -566,7 +566,7 @@ public More more() {
         Meter m = getOrCreateMeter(config, builder, id, mappedId, noopBuilder);
 
         if (!meterClass.isInstance(m)) {
-            throw new IllegalArgumentException("There is already a registered meter of a different type with the same name");
+            throw new IllegalArgumentException("There is already a registered meter of a different type with the same name: " + id.getName());
```

Comment:
Probably worth adding the types too.

Reply:
👍🏼, see: https://github.com/micrometer-metrics/micrometer/commit/64427f98396d07623be604b990af15ebf58e54c4